### PR TITLE
Moved tests out of run_in_graph_and_eager_mode in conditional gradient

### PR DESCRIPTION
**Pull request by autoih**
cc @gabrieldemarmiesse

### CODE REVIEW

File: tensorflow_addons/optimizers/conditional_gradient_test.py [hunk 1]
```diff
@@ -399,85 +399,78 @@ def test_tensor_learning_rate_and_conditional_gradient_nuclear():
...
+        return tf.math.reduce_sum(var0 + var1)
 
-        # the gradient for this loss function:
-        grads0 = tf.constant([[0, 0], [1, 1]], dtype=tf.float32)
-        norm0 = tf.math.reduce_sum(grads0 ** 2) ** 0.5
+    optimizer.minimize(loss, var_list=[var0, var1])
+    optimizer_variables = optimizer.variables()
+    # There should be three items. The first item is iteration,
+    # and one item for each variable.
+    assert "ConditionalGradient/var0" in optimizer_variables[1].name
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/optimizers/conditional_gradient_test.py:415]:
```suggestion
    assert optimizer_variables[1].name.startswith("ConditionalGradient/var0") 
```

The function startswith in python: https://docs.python.org/3/library/stdtypes.html#str.startswith

You can do a similar change to the others below.
